### PR TITLE
test: fix and add testing for verify-token predicate output

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Fixes https://github.com/slsa-framework/slsa-github-generator/issues/1541

* Updates the schedule verify-token action to use `output-predicate`.
* Adds predicate output testing for verify-token.

### CODE REVIEW

File: .github/actions/verify-token/dist/index.js [hunk 1]
```diff
@@ -88,8 +88,15 @@ function run() {
             const workflowRecipient = core.getInput("slsa-workflow-recipient");
             const unverifiedToken = core.getInput("slsa-unverified-token");
             const outputPredicate = core.getInput("output-predicate");
+            if (!outputPredicate) {
+                // detect if output predicate is null or empty string.
+                throw new Error("output-predicate must be supplied");
+            }
             const wd = (0, utils_1.getEnv)("GITHUB_WORKSPACE");
             const safeOutput = (0, utils_1.resolvePathInput)(outputPredicate, wd);
+            if (fs.existsSync(safeOutput)) {
```
Comment (laurentsimon) [inline .github/actions/verify-token/dist/index.js:97]: maybe you can create the function to address https://github.com/slsa-framework/slsa-github-generator/issues/1513 as part of the PR, like `safeWrite()`. Just a thought, not really necessary

File: .github/workflows/scripts/e2e-verify.common.sh [hunk 2]
```diff
@@ -56,3 +56,45 @@ e2e_verify_common_metadata() {
...
+    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "RUNNER_NAME" "$RUNNER_NAME"
+    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "RUNNER_OS" "$RUNNER_OS"
+}
+
+# Verifies common fields of the SLSA v1.0 predicate runDetails.
+# $1: the predicate content
+e2e_verify_common_runDetails_v1() {
+    # This does not include the builder ID since it is not common to all.
+    e2e_verify_predicate_v1_runDetails_metadata_invocationId "$1" "https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID/attempts/$GITHUB_RUN_ATTEMPT"
+}
```
Comment (laurentsimon) [inline .github/workflows/scripts/e2e-verify.common.sh:100]: nit: add the line

**Review by laurentsimon — APPROVED** [review]

File: .github/workflows/scripts/e2e-utils.sh [hunk 3]
```diff
@@ -75,3 +75,27 @@ _e2e_verify_query() {
...
+    _e2e_verify_query "$1" "$3" '.buildDefinition.systemParamters.'"$2"
+}
+
+e2e_verify_predicate_v1_runDetails_builder_id() {
+    _e2e_verify_query "$1" "$2" '.runDetails.builder.id'
+}
+
+e2e_verify_predicate_v1_runDetails_metadata_invocationId() {
+    _e2e_verify_query "$1" "$2" '.runDetails.metadata.invocationId'
+}
```
Comment (ianlewis) [inline .github/workflows/scripts/e2e-utils.sh:101]: nit: add newline here too

File: .github/workflows/scripts/e2e-verify.common.sh [hunk 4]
```diff
@@ -56,3 +56,45 @@ e2e_verify_common_metadata() {
 e2e_verify_common_materials() {
     e2e_verify_predicate_materials "$1" "{\"uri\":\"git+https://github.com/$GITHUB_REPOSITORY@$GITHUB_REF\",\"digest\":{\"sha1\":\"$GITHUB_SHA\"}}"
 }
+
+# Verifies common fields of the SLSA v1.0 predicate.
```
Comment (ianlewis) [inline .github/workflows/scripts/e2e-verify.common.sh:60]: Did we copy this file from example-package at some point? Any thoughts on how we could manage drift between the tests?

File: .github/workflows/scripts/schedule.actions/verify-token.sh [hunk 5]
```diff
@@ -49,3 +51,10 @@ e2e_assert_eq "$GITHUB_ACTOR" "$actor"
 e2e_assert_eq "$TOOL_REPOSITORY" "$GITHUB_REPOSITORY"
 e2e_assert_eq "$TOOL_REF" "$GITHUB_REF"
 e2e_assert_eq "$TOOL_URI" "https://github.com/$GITHUB_REPOSITORY/.github/workflows/e2e.verify-token.schedule.yml@$GITHUB_REF"
+
+PREDICATE_CONTENT=$(cat "$PREDICATE")
```
Comment (ianlewis) [inline .github/workflows/scripts/schedule.actions/verify-token.sh:55]: nit. The UUOC police might come and get you.
```suggestion
PREDICATE_CONTENT=$(<"$PREDICATE")
```

**Review by ianlewis — APPROVED** [review]

File: .github/actions/verify-token/dist/index.js (hunk 1 above)
Comment (asraa) [reply .github/actions/verify-token/dist/index.js:97]: Yes, I'm working on creating common utils now -- including a common file utility (resolving paths, safeWrite). I'll update that todo

**Pull request merged**